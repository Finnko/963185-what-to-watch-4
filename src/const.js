const Config = {
  MOVIE_MAX_RATING: 10,
  MOVIE_ACTORS_TO_SHOW: 4,
  SIMILAR_MOVIES_TO_SHOW: 4,
  MOVIES_NUMBER_TO_SHOW: 16,
  MOVIES_NUMBER_LIMIT: 8,
  MOVIES_NUMBER_OFFSET: 8,
  MAX_GENRES_TO_SHOW: 9,
  MOVIE_RATING_MAP: {
    0: `Bad`,
    3: `Normal`,
    5: `Good`,
    8: `Very good`,
    10: `Awesome`
  },
  ERRORS: {
    FETCH_DATA: `Something went wrong, please try again later.`,
  },
  LOADER: {
    SMALL: `small`,
    MEDIUM: `medium`,
  },
  DEFAULT_FILTER: `All genres`,
};

const PathName = {
  ROOT: `/`,
  MOVIE_PAGE: `/movie/`,
};

const TabName = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`
};

const ViewMode = {
  PROMO: {
    MAIN: `MAIN`,
    DETAILS: `DETAILS`
  },
  MOVIE_CARD: {
    WITH_PLAYER: `WITH_PLAYER`,
    IMAGE: `IMAGE`,
  }
};

export {
  Config,
  PathName,
  TabName,
  ViewMode,
};
