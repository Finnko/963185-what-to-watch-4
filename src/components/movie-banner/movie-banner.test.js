import React from 'react';
import renderer from 'react-test-renderer';
import {createMemoryHistory} from 'history';
import {Router} from 'react-router-dom';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import NameSpace from '../../store/name-space';
import {AuthStatus} from '../../const.ts';
import MovieBanner from './movie-banner.jsx';

const mock = {
  id: 1,
  title: `The Grand Budapest Hotel`,
  genre: `Drama`,
  thumb: `img/bohemian-rhapsody.jpg`,
  releaseYear: 2014,
  preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  backgroundImage: `img/bg-the-grand-budapest-hotel.jpg`,
  description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
  director: `Michael Bay`,
  actors: [`Leonardo Di Caprio`],
  rating: 7.5,
  ratingCount: 250,
  isFavorite: false,
  poster: `/img/the-grand-budapest-hotel-poster.jpg`,
};
const mockUser = {
  id: 1,
  email: `Oliver.conner@gmail.com`,
  name: `Oliver.conner`,
  avatarUrl: `img/1.png`
};

const props = {
  movie: mock,
  isAuth: true,
  updateFavoriteStatus: () => {},
};

const mockStore = configureStore([]);

const store = mockStore({
  [NameSpace.USER]: {
    user: mockUser,
    authStatus: AuthStatus.NO_AUTH,
  }
});

describe(`MovieBanner component render correctly`, () => {
  it(`Should MovieBanner component render correctly with main view`, () => {
    const history = createMemoryHistory();

    const tree = renderer
      .create(
          <Provider store={store}>
            <Router history={history}>
              <MovieBanner
                {...props}
                loading={false}
                error={false}
                viewMode={`MAIN`}
              />
            </Router>
          </Provider>
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should MovieBanner component render correctly with details view`, () => {
    const history = createMemoryHistory();

    const tree = renderer
      .create(
          <Provider store={store}>
            <Router history={history}>
              <MovieBanner
                {...props}
                loading={false}
                error={false}
                viewMode={`DETAILS`}
              />
            </Router>
          </Provider>
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should MovieBanner component render correctly with loading`, () => {
    const history = createMemoryHistory();

    const tree = renderer
      .create(
          <Provider store={store}>
            <Router history={history}>
              <MovieBanner
                {...props}
                loading={true}
                error={false}
                viewMode={`DETAILS`}
              />
            </Router>
          </Provider>
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should MovieBanner component render correctly with error`, () => {
    const history = createMemoryHistory();

    const tree = renderer
      .create(
          <Provider store={store}>
            <Router history={history}>
              <MovieBanner
                {...props}
                loading={false}
                error={true}
                viewMode={`DETAILS`}
              />
            </Router>
          </Provider>
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
