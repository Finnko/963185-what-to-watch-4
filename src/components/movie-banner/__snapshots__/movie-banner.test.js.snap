// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieBanner component render correctly Should MovieBanner component render correctly with details view 1`] = `
Array [
  <div
    className="movie-card__bg"
  >
    <img
      alt="The Grand Budapest Hotel"
      src="img/bg-the-grand-budapest-hotel.jpg"
    />
  </div>,
  <h1
    className="visually-hidden"
  >
    WTW
  </h1>,
  <header
    className="page-header movie-card__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="user-block"
    >
      <a
        className="user-block__link"
        href="/login"
        onClick={[Function]}
      >
        Sign In
      </a>
    </div>
  </header>,
  <div
    className="movie-card__wrap"
  >
    <div
      className=""
    >
      <div
        className="visually-hidden"
      >
        <img
          alt="The Grand Budapest Hotel"
          height="327"
          src="/img/the-grand-budapest-hotel-poster.jpg"
          width="218"
        />
      </div>
      <div
        className="movie-card__desc"
      >
        <h2
          className="movie-card__title"
        >
          The Grand Budapest Hotel
        </h2>
        <p
          className="movie-card__meta"
        >
          <span
            className="movie-card__genre"
          >
            Drama
          </span>
          <span
            className="movie-card__year"
          >
            2014
          </span>
        </p>
        <div
          className="movie-card__buttons"
        >
          <a
            className="btn btn--play movie-card__button"
            href="/player/1"
            onClick={[Function]}
            type="button"
          >
            <svg
              height="19"
              viewBox="0 0 19 19"
              width="19"
            >
              <use
                xlinkHref="#play-s"
              />
            </svg>
            <span>
              Play
            </span>
          </a>
          <button
            className="btn btn--list movie-card__button"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <svg
              height="20"
              viewBox="0 0 19 20"
              width="19"
            >
              <use
                xlinkHref="#add"
              />
            </svg>
            <span>
              My list
            </span>
          </button>
          <a
            className="btn movie-card__button"
            href="/films/1/review"
            onClick={[Function]}
          >
            Add review
          </a>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`MovieBanner component render correctly Should MovieBanner component render correctly with error 1`] = `
Array [
  <div
    className="movie-card__bg"
  >
    <img
      alt="The Grand Budapest Hotel"
      src="img/bg-the-grand-budapest-hotel.jpg"
    />
  </div>,
  <h1
    className="visually-hidden"
  >
    WTW
  </h1>,
  <header
    className="page-header movie-card__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="user-block"
    >
      <a
        className="user-block__link"
        href="/login"
        onClick={[Function]}
      >
        Sign In
      </a>
    </div>
  </header>,
  <div
    className="movie-card__wrap"
  >
    <div
      className=""
    >
      <div
        className="visually-hidden"
      >
        <img
          alt="The Grand Budapest Hotel"
          height="327"
          src="/img/the-grand-budapest-hotel-poster.jpg"
          width="218"
        />
      </div>
      <div
        className="movie-card__desc"
      >
        <h2
          className="movie-card__title"
        >
          The Grand Budapest Hotel
        </h2>
        <p
          className="movie-card__meta"
        >
          <span
            className="movie-card__genre"
          >
            Drama
          </span>
          <span
            className="movie-card__year"
          >
            2014
          </span>
        </p>
        <div
          className="movie-card__buttons"
        >
          <a
            className="btn btn--play movie-card__button"
            href="/player/1"
            onClick={[Function]}
            type="button"
          >
            <svg
              height="19"
              viewBox="0 0 19 19"
              width="19"
            >
              <use
                xlinkHref="#play-s"
              />
            </svg>
            <span>
              Play
            </span>
          </a>
          <button
            className="btn btn--list movie-card__button"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <svg
              height="20"
              viewBox="0 0 19 20"
              width="19"
            >
              <use
                xlinkHref="#add"
              />
            </svg>
            <span>
              My list
            </span>
          </button>
          Ooops, please try again.
          <a
            className="btn movie-card__button"
            href="/films/1/review"
            onClick={[Function]}
          >
            Add review
          </a>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`MovieBanner component render correctly Should MovieBanner component render correctly with loading 1`] = `
Array [
  <div
    className="movie-card__bg"
  >
    <img
      alt="The Grand Budapest Hotel"
      src="img/bg-the-grand-budapest-hotel.jpg"
    />
  </div>,
  <h1
    className="visually-hidden"
  >
    WTW
  </h1>,
  <header
    className="page-header movie-card__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="user-block"
    >
      <a
        className="user-block__link"
        href="/login"
        onClick={[Function]}
      >
        Sign In
      </a>
    </div>
  </header>,
  <div
    className="movie-card__wrap"
  >
    <div
      className=""
    >
      <div
        className="visually-hidden"
      >
        <img
          alt="The Grand Budapest Hotel"
          height="327"
          src="/img/the-grand-budapest-hotel-poster.jpg"
          width="218"
        />
      </div>
      <div
        className="movie-card__desc"
      >
        <h2
          className="movie-card__title"
        >
          The Grand Budapest Hotel
        </h2>
        <p
          className="movie-card__meta"
        >
          <span
            className="movie-card__genre"
          >
            Drama
          </span>
          <span
            className="movie-card__year"
          >
            2014
          </span>
        </p>
        <div
          className="movie-card__buttons"
        >
          <a
            className="btn btn--play movie-card__button"
            href="/player/1"
            onClick={[Function]}
            type="button"
          >
            <svg
              height="19"
              viewBox="0 0 19 19"
              width="19"
            >
              <use
                xlinkHref="#play-s"
              />
            </svg>
            <span>
              Play
            </span>
          </a>
          <button
            className="btn btn--list movie-card__button"
            disabled={true}
            onClick={[Function]}
            type="button"
          >
            <svg
              height="20"
              viewBox="0 0 19 20"
              width="19"
            >
              <use
                xlinkHref="#add"
              />
            </svg>
            <span>
              My list
            </span>
          </button>
          <a
            className="btn movie-card__button"
            href="/films/1/review"
            onClick={[Function]}
          >
            Add review
          </a>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`MovieBanner component render correctly Should MovieBanner component render correctly with main view 1`] = `
Array [
  <div
    className="movie-card__bg"
  >
    <img
      alt="The Grand Budapest Hotel"
      src="img/bg-the-grand-budapest-hotel.jpg"
    />
  </div>,
  <h1
    className="visually-hidden"
  >
    WTW
  </h1>,
  <header
    className="page-header movie-card__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="user-block"
    >
      <a
        className="user-block__link"
        href="/login"
        onClick={[Function]}
      >
        Sign In
      </a>
    </div>
  </header>,
  <div
    className="movie-card__wrap"
  >
    <div
      className="movie-card__info"
    >
      <div
        className="movie-card__poster"
      >
        <img
          alt="The Grand Budapest Hotel"
          height="327"
          src="/img/the-grand-budapest-hotel-poster.jpg"
          width="218"
        />
      </div>
      <div
        className="movie-card__desc"
      >
        <h2
          className="movie-card__title"
        >
          The Grand Budapest Hotel
        </h2>
        <p
          className="movie-card__meta"
        >
          <span
            className="movie-card__genre"
          >
            Drama
          </span>
          <span
            className="movie-card__year"
          >
            2014
          </span>
        </p>
        <div
          className="movie-card__buttons"
        >
          <a
            className="btn btn--play movie-card__button"
            href="/player/1"
            onClick={[Function]}
            type="button"
          >
            <svg
              height="19"
              viewBox="0 0 19 19"
              width="19"
            >
              <use
                xlinkHref="#play-s"
              />
            </svg>
            <span>
              Play
            </span>
          </a>
          <button
            className="btn btn--list movie-card__button"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <svg
              height="20"
              viewBox="0 0 19 20"
              width="19"
            >
              <use
                xlinkHref="#add"
              />
            </svg>
            <span>
              My list
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>,
]
`;
